distinctipy/color.py

```diff
@@ -0,0 +1,167 @@
+from typing import Tuple, List
+from enum import IntEnum
+from numbers import Number
+
+from .distinctipy import *
+
+
+class ColorOrder(IntEnum):
+    RGB = 1  # Default of the library
+    BGR = 2
+
+
+class DataType(IntEnum):
+    FLOAT = 1  # Default of the library, 0..1 float
+    INT8 = 2  # Used by cv2, 0..255 int
+    INT16 = 3  # 0..65535 int
+
+
+class Color:
+    colors: Tuple[Number, Number, Number]
+
+    _color_order: ColorOrder = ColorOrder.RGB
+    _data_type: DataType = DataType.FLOAT
+
+    def __init__(self, colors: Tuple[Number, Number, Number],
+                 color_order: ColorOrder = ColorOrder.RGB,
+                 data_type: DataType = DataType.FLOAT):
+
+        self.colors = colors
+        self._color_order = color_order
+        self._data_type = data_type
+
+    @staticmethod
```

Comment:
Minor comment but would `@classmethod` be more appropriate for all the functions that return a `Color` instance?

Reply:
@jack89roberts I am not that familiar with Python, so I had to look this up. I guess it would be useful, but I have a concern that "Class methods can be can be called from instances and from **the class itself**" - as these methods can't be called from the class itself, as they have empty variables (`colors`, `_data_type`, `_color_order`). Thoughts?

Reply:
I don't quite understand what you mean by the empty variables, but one of the functions could become something like

```python
@classmethod
def from_rgb8(cls, rgb8: Tuple[int, int, int]) -> 'Color':
    return cls(rgb8, color_order=ColorOrder.RGB, data_type=DataType.INT8)
```

which could be called as `Color.from_rgb8((255, 0, 127))` or `color_instance..from_rgb8((255, 0, 127))`. I think both `@classmethod` and `@staticmethod` would function the same here, essentially, but this is what `@classmethod` is intended for.

Reply:
Thanks for the elaboration! I am still not convinced when it would be useful to create a new Color object from the Color instance, could you give an example?

Reply:
Calling from the instance isn't really useful like you say (you could also do that with staticmethod). The main advantage, as I understand it, is if you need to access other properties/functions of the class (not the instance) and/or to take advantage of inheritance. E.g.

```python
class Animal:
    def __init__(self, name):
        self.name = name

    @classmethod
    def default(cls):
        return cls("Foo")


class Dog(Animal):
    def bark(self):
        print("Woof!")


d = Dog.default()
print(type(d))  # <class '__main__.Dog'>
print(d.name)  # 'Foo'
```

`d` is a `Dog`instance, not an `Animal` instance (as `cls` in `default(cls)` resolves to the class of caller).

Reply:
Thank you, just updated it:)